### DISCUSSION TITLE
Support for terraform private registry

### DISCUSSION THREAD

**Discussion by mkjmdski** [renovatebot/renovate#11878] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### Please select which platform you are using if self-hosting.

gitlab.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

The newest according to the template for gitlab.ci job 

### Describe the bug

Docs describe that terraform registry is supported for both private and public. However there is no information what credentials need to be provided to update modules hosted on private registry.

The following message is displayed for code that contains private registry module:

```shell
Please correct - or verify that you can safely ignore - these lookup failures before you merge this MR.
Failed to look up dependency app.terraform.io/toolkitz/project-variables/gitlab
```

Is there anyway to configure that with some kind of token?

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste any log here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by viceice** [comment]
What header needs to be set to grant access to the private registry? is it a normal `Authentication: Bearer <token>` ?

**Comment by viceice** [comment, marked as answer]
Normally you simply need to add a host rule with `hostType=<datasource-id>`

https://github.com/renovatebot/renovate/blob/a6f4e99e2796d0d2d38c466b35d3a7293bfc386f/lib/datasource/terraform-module/index.ts#L11-L11

Maybe this will work
```js
module.exports = {
  hostRules: [
    {
      hostType: 'terraform-module',
      matchHost: 'app.terraform.io',
      token: process.env.TERRA_TOKEN,
    },
}
```

**Reply by ryancausey** [reply]
Should this be working automatically for gitlab.com hosted private Terraform module registries? I had to set a similar host rule but for 'gitlab.com' instead of 'app.terraform.io' in order to get it to work for me when I thought the `RENOVATE_TOKEN` would be used directly.

**Comment by secustor** [comment]
Yes, it is a bearer token and the example above should work without a problem.    
https://www.terraform.io/docs/cloud/users-teams-organizations/users.html#api-tokens

**Reply by viceice** [reply]
Thanks, converted to discussion, as there is no bug 🙃
